src/hooks/useSlider.js

```diff
@@ -3,26 +3,26 @@ import { useSpring } from 'react-spring'
 import { useDrag } from 'react-use-gesture'
 import { clamp } from '../helpers'
 
-export default function useSlider({ slides }) {
+export default function useSlider({ initialSlide, slides }) {
+  const slideIndex = clamp(initialSlide, 0, slides.length - 1)
+  const sliderX = typeof window === 'object' ? -window.innerWidth * slideIndex : 0
+
   const [{ x, scale }, set] = useSpring(() => ({
-    x: 0,
+    x: sliderX,
     scale: 1,
     config: { tension: 270, clamp: true },
   }))
-  const index = useRef(0)
+
+  const index = useRef(slideIndex)
 
   // Slide numbers (for display purposes only)
-  const [currentSlide, updateSlide] = useState(0)
+  const [currentSlide, updateSlide] = useState(slideIndex)
   const [zooming, setZooming] = useState(false)
 
   const onScale = useCallback(
     slideProps => {
       set({ scale: slideProps.scale })
-      if (slideProps.scale === 1) {
-        setZooming(false)
-      } else {
-        setZooming(true)
-      }
+      setZooming(slideProps.scale === 1)
```

Comment:
This is the inverse of the existing logic - when the scale is 1, it means the element is not being zoomed in on.

The equivalent would be:

```js
setZooming(slideProps.scale !== 1)
```

With that being said, I think the shorthand reduces readability due to the inverted expression (i.e. `scale === 1` vs `scale !== 1`).